draco/util/random.py

```diff
@@ -0,0 +1,330 @@
+"""Utilities for drawing random numbers.
+
+Routines
+========
+
+.. autosummary::
+    :toctree:
+
+    standard_complex_wishart
+    draw_complex_wishart
+    mpi_random_seed
+"""
+import contextlib
+
+import numpy as np
+import zlib
+
+from caput import config
+from ..core import task
+
+
+_rng = None
+_default_bitgen = np.random.SFC64
+
+
+def default_rng():
+    """Returns an instance of the default random number generator to use.
+
+    This creates a randomly seeded generator using the fast SFC64 bit generator
+    underneath. This is only initialise on the first call, subsequent calls will
+    return the same Generator.
+
+    Returns
+    -------
+    rng : np.random.Generator
+    """
+
+    global _rng
+
+    if _rng is None:
+        _rng = np.random.Generator(_default_bitgen())
+
+    return _rng
+
+
+def complex_normal(
+    size=None, loc=0.0, scale=1.0, dtype=np.complex128, rng=None, out=None
+):
+    """Get a set of complex normal variables.
+
+        By default generate standard complex normal variables.
+
+        Parameters
+        ----------
+        size : tuple
+            Shape of the array of variables.
+        loc : float, optional
```

Comment:
May want to indicate in doc string that `loc` and `scale` can be any `np.ndarray` that broadcasts with an array of shape `size`, as is being done in the `GaussianNoise` and `GaussianNoiseDataset` task.